Tests/WebAuthnTests/WebAuthnManagerTests.swift

```diff
@@ -345,20 +302,51 @@ final class WebAuthnManagerTests: XCTestCase {
+        let credentialPublicKey: [UInt8] = TestCredentialPublicKeyBuilder().validMock().buildAsByteArray()
+        let authData = TestAuthDataBuilder()
+            .validMock()
+            .attestedCredData(credentialPublicKey: credentialPublicKey)
+            .noExtensionData()
+        let attestationObject = TestAttestationObjectBuilder()
+            .validMock()
+            .authData(authData)
+            .buildBase64URLEncoded()
+        let credential = try await finishRegistration(id: credentialID, attestationObject: attestationObject)
+        XCTAssertNotNil(credential)
+
+        XCTAssertEqual(credential.id, credentialID.asString())
+        XCTAssertEqual(credential.publicKey, credentialPublicKey)
+    }
+
+    // Swift CBOR library currently crashes when running this test. WE NEED TO FIX THIS
+    // TODO: Fix this test
+    // func testFinishRegistrationFuzzying() async throws {
+    //     for _ in 1...50 {
+    //         let length = Int.random(in: 1...10_000_000)
+    //         let randomAttestationObject: URLEncodedBase64 = Data(
+    //             [UInt8](repeating: UInt8.random(), count: length)
+    //         ).base64URLEncodedString()
+    //         let shouldBeNil = try? await finishRegistration(attestationObject: randomAttestationObject)
+    //         XCTAssertNil(shouldBeNil)
+    //     }
+    // }
+
     private func finishRegistration(
-        challenge: EncodedBase64 = "cmFuZG9tU3RyaW5nRnJvbVNlcnZlcg",
-        id: String = "4PrJNQUJ9xdI2DeCzK9rTBRixhXHDiVdoTROQIh8j80",
+        challenge: EncodedBase64 = "cmFuZG9tU3RyaW5nRnJvbVNlcnZlcg", // "randomStringFromServer"
+        id: EncodedBase64 = "4PrJNQUJ9xdI2DeCzK9rTBRixhXHDiVdoTROQIh8j80",
         type: String = "public-key",
         rawID: URLEncodedBase64 = "4PrJNQUJ9xdI2DeCzK9rTBRixhXHDiVdoTROQIh8j80",
-        clientDataJSON: URLEncodedBase64 = "eyJ0eXBlIjoid2ViYXV0aG4uY3JlYXRlIiwiY2hhbGxlbmdlIjoiY21GdVpHOXRVM1J5YVc1blJuSnZiVk5sY25abGNnIiwib3JpZ2luIjoiaHR0cHM6Ly9leGFtcGxlLmNvbSIsImNyb3NzT3JpZ2luIjpmYWxzZSwib3RoZXJfa2V5c19jYW5fYmVfYWRkZWRfaGVyZSI6ImRvIG5vdCBjb21wYXJlIGNsaWVudERhdGFKU09OIGFnYWluc3QgYSB0ZW1wbGF0ZS4gU2VlIGh0dHBzOi8vZ29vLmdsL3lhYlBleCJ9",
-        attestationObject: URLEncodedBase64 = "o2NmbXRkbm9uZWdhdHRTdG10oGhhdXRoRGF0YVg5o3mm9u6vuaVeN4wRgDTidR5oL6ufLTCrE9ISVYbOGUdBAAAAAKN5pvbur7mlXjeMEYA04nUAAQAA",
+        clientDataJSON: URLEncodedBase64 = TestClientDataJSON().base64URLEncoded,
+        attestationObject: URLEncodedBase64 = TestAttestationObjectBuilder().validMock().buildBase64URLEncoded(),
         requireUserVerification: Bool = false,
         confirmCredentialIDNotRegisteredYet: (String) async throws -> Bool = { _ in true }
     ) async throws -> Credential {
         try await webAuthnManager.finishRegistration(
             challenge: challenge,
             credentialCreationData: RegistrationCredential(
-                id: id,
+                id: id.asString(),
```

Comment:
Is it worth extracting this to it's own type instead of just a `String`?

Reply:
We're handling a lot of base64/base64url data and having some base64 "type safety" is super helpful.